# Project X -- [Prisha] [Joshi]


**TA Help:** John Smith, Alice Jones

- Help with figuring out how to write a function.
    
**Collaboration:** Friend1, Friend2
    
- Helped figuring out how to load the dataset.
- Helped debug error with my plot.

## Question 1

In [1]:
# code here
library(data.table)
myDF <- fread("/anvil/projects/tdm/data/death_records/DeathRecords.csv")

sub <- subset(myDF, Sex == 'F')
head(sub)
table(sub$Age,useNA = 'always')
sub1 <- subset(myDF, (Sex =='F') & (Age!=999))
table(sub1$Age)

Id ResidentStatus Education1989Revision Education2003Revision
1  3 1              0                     7                    
2  6 1              0                     5                    
3  9 1              0                     3                    
4 11 1              0                     3                    
5 13 1              0                     4                    
6 14 1              0                     3                    
  EducationReportingFlag MonthOfDeath Sex AgeType Age AgeSubstitutionFlag ...
1 1                      1            F   1       75  0                   ...
2 1                      1            F   1       93  0                   ...
3 1                      1            F   1       86  0                   ...
4 1                      1            F   1       79  0                   ...
5 1                      1            F   1       85  0                   ...
6 1                      1            F   1       84  0                   ...
  CauseRecode39 NumberOfEntityAxisConditions NumberOfRecordAxisConditions Race
1 28            2                            2                            1   
2 37            5                            5                            1   
3 37            1                            1                            1   
4 22            2                            2                            1   
5 22            5                            5                            1   
6  8            2                            2                            1   
  BridgedRaceFlag RaceImputationFlag RaceRecode3 RaceRecode5 HispanicOrigin
1 0               0                  1           1           100           
2 0               0                  1           1           100           
3 0               0                  1           1           100           
4 0               0                  1           1           100           
5 0               0                  1           1           100           
6 0               0                  1           1           100           
  HispanicOriginRaceRecode
1 6                       
2 6                       
3 6                       
4 6                       
5 6                       
6 6


    1     2     3     4     5     6     7     8     9    10    11    12    13 
 3164  2100  1346   937   820   583   495   511   418   472   358   347   386 
   14    15    16    17    18    19    20    21    22    23    24    25    26 
  410   499   533   636   735   827   936  1058  1117  1086  1155  1200  1320 
   27    28    29    30    31    32    33    34    35    36    37    38    39 
 1264  1340  1438  1452  1535  1683  1804  1878  2059  1999  2226  2267  2413 
   40    41    42    43    44    45    46    47    48    49    50    51    52 
 2577  2953  3324  3681  4091  4229  4524  4957  5664  6520  7247  8190  8740 
   53    54    55    56    57    58    59    60    61    62    63    64    65 
 9616 10189 11062 11748 12443 12762 13358 13977 14518 15142 15664 16284 17276 
   66    67    68    69    70    71    72    73    74    75    76    77    78 
18813 20865 17669 18584 20486 22755 22041 22074 22656 24026 25305 26091 27780 
   79    80    81    82    83    84    85    86    


    1     2     3     4     5     6     7     8     9    10    11    12    13 
 3164  2100  1346   937   820   583   495   511   418   472   358   347   386 
   14    15    16    17    18    19    20    21    22    23    24    25    26 
  410   499   533   636   735   827   936  1058  1117  1086  1155  1200  1320 
   27    28    29    30    31    32    33    34    35    36    37    38    39 
 1264  1340  1438  1452  1535  1683  1804  1878  2059  1999  2226  2267  2413 
   40    41    42    43    44    45    46    47    48    49    50    51    52 
 2577  2953  3324  3681  4091  4229  4524  4957  5664  6520  7247  8190  8740 
   53    54    55    56    57    58    59    60    61    62    63    64    65 
 9616 10189 11062 11748 12443 12762 13358 13977 14518 15142 15664 16284 17276 
   66    67    68    69    70    71    72    73    74    75    76    77    78 
18813 20865 17669 18584 20486 22755 22041 22074 22656 24026 25305 26091 27780 
   79    80    81    82    83    84    85    86    

the deliverables are seen above.the head of the subset is visible that only has female data. the subset with age is also shown with and without the missing values. 

## Question 2

In [2]:
# code here
library(data.table)
df <- fread("/anvil/projects/tdm/data/8451/The_Complete_Journey_2_Master/5000_transactions.csv")
refundsDF <- subset(df, SPEND <0 )
table(refundsDF$STORE_R)


CENTRAL    EAST   SOUTH    WEST 
   2750    3269    2675    3952 

central store has 2750 refunds. east store has 3269 refunds. south store has 2675 refunds and west store has 3952 refunds. 

## Question 3

In [7]:
# code here
library(data.table)
df2 <- fread("/anvil/projects/tdm/data/beer/reviews_sample.csv")
sub3 <- subset(df2, (score != 5) & (overall == 5))
nrow(sub3)

[1] 12436

12436 lines of data have the score less than 5 or the overall score exactly equal to 5 

## Question 4

In [4]:
# code here
library(data.table)
myDF <- fread("/anvil/projects/tdm/data/election/itcont1980.txt", quote="")
names(myDF) <- c("CMTE_ID", "AMNDT_IND", "RPT_TP", "TRANSACTION_PGI", "IMAGE_NUM", "TRANSACTION_TP", "ENTITY_TP", "NAME", "CITY", "STATE", "ZIP_CODE", "EMPLOYER", "OCCUPATION", "TRANSACTION_DT", "TRANSACTION_AMT", "OTHER_ID", "TRAN_ID", "FILE_NUM", "MEMO_CD", "MEMO_TEXT", "SUB_ID")
missing_data <- myDF[is.na(TRANSACTION_DT),]
donation <- missing_data[, .N, by = NAME]
six_donations <- donation[N==6,]
print(six_donations)

                      NAME     N
                    <char> <int>
1: SCHECK, RICHARD HERBERT     6


the person who made six donations in 1980 was richard herbet scheck. 

## Question 5

In [1]:
# code here
library(data.table)
df3 <- fread("/anvil/projects/tdm/data/flights/subset/1990.csv")
missing <- df3[is.na(DepDelay) & is.na(ArrDelay),.N]
missing
dep_miss <- df3[is.na(DepDelay) & !is.na(ArrDelay),.N]
dep_miss
arr_miss <- df3[is.na(ArrDelay) & !is.na(DepDelay),.N]
arr_miss
                


[1] 52458

[1] 0

[1] 15954

the number of flights where the arrival and departure data was missing  was 52458. the number of flights where only the departure delay was mssing was 0 and the number of flights where only the arrival was missing was 15954 

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.